File: src/WellTool.Cron/hutool-core/src/main/java/cn/hutool/core/bean/copier/provider/MapValueProvider.java
package cn.hutool.core.bean.copier.provider;

import cn.hutool.core.bean.copier.ValueProvider;
import cn.hutool.core.convert.Convert;

import java.lang.reflect.Type;
import java.util.Map;

/**
 * Map值提供者
 *
 * @author Looly
 * @since 5.8.40
 */
@SuppressWarnings("rawtypes")
public class MapValueProvider implements ValueProvider<String> {

	private final Map map;

	/**
	 * 构造
	 *
	 * @param map map
	 */
	public MapValueProvider(final Map map) {
		this.map = map;
	}

	@Override
	public Object value(String key, Type valueType) {
		return Convert.convert(valueType, map.get(key));
	}

	@Override
	public boolean containsKey(String key) {
		return map.containsKey(key);
	}
}

File: src/WellTool.Core/Bean/Copier/MapValueProvider.cs
using System;
using System.Collections;

namespace WellTool.Core.Bean.Copier
{
    /// <summary>
    /// 字典值提供者
    /// </summary>
    public class MapValueProvider : IValueProvider<string>
    {
        private readonly IDictionary _map;

        /// <summary>
        /// 构造函数
        /// </summary>
        /// <param name="map">字典对象</param>
        public MapValueProvider(IDictionary map)
        {
            _map = map;
        }

        /// <summary>
        /// 获取值
        /// </summary>
        public object Value(string key, Type valueType)
        {
            if (_map.Contains(key))
            {
                return _map[key];
            }
            return null;
        }

        /// <summary>
        /// 是否包含指定属性
        /// </summary>
        public bool ContainsKey(string key)
        {
            return _map.Contains(key);
        }
    }
}
